barretenberg/cpp/src/barretenberg/vm2/tracegen/precomputed_trace.cpp

```diff
@@ -215,6 +216,82 @@ void PrecomputedTraceBuilder::process_wire_instruction_spec(TraceContainer& trac
+    // Memory Access for Registers
+    trace.reserve_column(C::precomputed_mem_op_reg1, num_opcodes);
+    trace.reserve_column(C::precomputed_rw_1, num_opcodes);
+    trace.reserve_column(C::precomputed_mem_op_reg2, num_opcodes);
+    trace.reserve_column(C::precomputed_rw_2, num_opcodes);
+    trace.reserve_column(C::precomputed_mem_op_reg3, num_opcodes);
+    trace.reserve_column(C::precomputed_rw_3, num_opcodes);
+    trace.reserve_column(C::precomputed_mem_op_reg4, num_opcodes);
+    trace.reserve_column(C::precomputed_rw_4, num_opcodes);
+    trace.reserve_column(C::precomputed_mem_op_reg5, num_opcodes);
+    trace.reserve_column(C::precomputed_rw_5, num_opcodes);
+    trace.reserve_column(C::precomputed_mem_op_reg6, num_opcodes);
+    trace.reserve_column(C::precomputed_rw_6, num_opcodes);
+    trace.reserve_column(C::precomputed_mem_op_reg7, num_opcodes);
+    trace.reserve_column(C::precomputed_rw_7, num_opcodes);
+    // Gadget/Subtrace Selectors
+    trace.reserve_column(C::precomputed_op_alu_sel, num_opcodes);
+    trace.reserve_column(C::precomputed_op_bitwise_sel, num_opcodes);
+    trace.reserve_column(C::precomputed_op_poseidon_perm_sel, num_opcodes);
+    trace.reserve_column(C::precomputed_op_to_radix_sel, num_opcodes);
+    trace.reserve_column(C::precomputed_subtrace_operation_id, num_opcodes);
+
+    uint32_t row = 0;
+    for (const auto& [exec_opcode, exec_instruction_spec] : EXEC_INSTRUCTION_SPEC) {
+        auto dispatch_to_subtrace = SUBTRACE_INFO_MAP.at(exec_opcode);
+        bool alu_sel = dispatch_to_subtrace.subtrace_selector == SubtraceSel::ALU;
+        bool bitwise_sel = dispatch_to_subtrace.subtrace_selector == SubtraceSel::BITWISE;
+        bool poseidon_sel = dispatch_to_subtrace.subtrace_selector == SubtraceSel::POSEIDON2PERM;
+        bool to_radix_sel = dispatch_to_subtrace.subtrace_selector == SubtraceSel::TORADIXBE;
+        bool ecc_sel = dispatch_to_subtrace.subtrace_selector == SubtraceSel::ECC;
+
+        auto register_info = REGISTER_INFO_MAP.at(exec_opcode);
+
+        trace.set(row,
+                  { { { C::precomputed_exec_opcode_value, static_cast<uint32_t>(exec_opcode) },
+                      { C::precomputed_exec_opcode_base_l2_gas, exec_instruction_spec.gas_cost.base_l2 },
+                      { C::precomputed_exec_opcode_base_da_gas, exec_instruction_spec.gas_cost.base_da },
+                      { C::precomputed_exec_opcode_dynamic_l2_gas, exec_instruction_spec.gas_cost.dyn_l2 },
+                      { C::precomputed_exec_opcode_dynamic_da_gas, exec_instruction_spec.gas_cost.dyn_da },
+                      // Memory Access for registers
+                      { C::precomputed_mem_op_reg1, register_info & 1 },
+                      { C::precomputed_rw_1, register_info >> 1 & 1 },
+                      { C::precomputed_mem_op_reg2, (register_info >> 2) & 1 },
+                      { C::precomputed_rw_2, (register_info >> 3) & 1 },
```

Comment:
why not just store the actual registerinfo struct and do something like `.write()`?

Reply:
let's discuss so i have a better idea what you're suggesting